server/filestore.go

```diff
@@ -9468,6 +9460,15 @@ func (o *consumerFileStore) Update(state *ConsumerState) error {
 		}
 	}
 
+	// Replace our state.
+	o.mu.Lock()
+	defer o.mu.Unlock()
+
+	// Check to see if this is an outdated update.
+	if state.Delivered.Consumer < o.state.Delivered.Consumer || state.AckFloor.Stream < o.state.AckFloor.Stream {
+		return fmt.Errorf("old update ignored")
```

Comment:
What effect upstream does this have by changing this to an error where as before it was ignored silently?

Reply:
A few cases where this has an effect upstream, most importantly all places already handle/bubble up any error. This is now also more consistent, since it would be done for memstore already but not for filestore before this change.

Also, `setStoreState` improves in that it doesn't call `o.applyState(state)` when the `o.store.Update(state)` was actually ignored/errored. Which was a bug before that would have the states drift.
```go
	err := o.store.Update(state)
	if err == nil {
		o.applyState(state)
	}
```